File: POG/binarytodecimal.c
#include <stdio.h>    
#include <math.h> // use math.h header file  
void main()  
{  
    // declaration of local variables i, bin_num, decimal_num = 0, rem;  
    int i, bin_num, decimal_num = 0, rem;  
    printf (" Enter the binary number with the combination of 0s and 1s \n");  
    scanf ("%d", &bin_num); // takes the binary number as the input  
      
    printf( " \n The binary number is %d", bin_num); // print the binary number  
    for (i = 0; bin_num != 0; ++i)  
    {  
        rem = bin_num % 10;  
        bin_num = bin_num / 10;  
        decimal_num = decimal_num + (rem) * ( pow (2, i));  
          
    }  
    // print the decimal number  
    printf ("\n Conversion from binary to decimal number is %d", decimal_num);  
   
}  

File: Core Java/Bin_Dec_Conversion/src/study/Binary_To_Decimal.java
package study;

public class Binary_To_Decimal {

	public static void main(String[] args) {

		System.out.println(convertToDecimal(100));
	}

	private static int  convertToDecimal(int binary) {
		int decimal=0;
		int i=0;

		while(true)
		{
			if(binary==0)
			{
				break;
			}
			else {
					int temp=binary%10;
					decimal+=temp*Math.pow(2,i);
					binary=binary/10;
					i++;
			    }
		}
		return decimal;


	}

}
